lxd/network/driver_common.go

```diff
@@ -578,11 +578,23 @@ func (n *common) handleDependencyChange(netName string, netConfig map[string]str
 func (n *common) bgpValidationRules(config map[string]string) (map[string]func(value string) error, error) {
 	rules := map[string]func(value string) error{}
 	for k := range config {
-		// BGP keys have the peer name in their name, extract the suffix.
+		// only BGP keys are handled here.
 		if !strings.HasPrefix(k, "bgp.") {
 			continue
 		}
 
+		switch k {
+		case "bgp.ipv4.nexthop":
+			rules[k] = validate.Optional(validate.IsNetworkAddressV4)
+		case "bgp.ipv6.nexthop":
+			rules[k] = validate.Optional(validate.IsNetworkAddressV6)
+		}
+
+		// BGP keys have the peer name in their name, extract the suffix.
+		if !strings.HasPrefix(k, "bgp.peers.") {
```

Comment:
Probably we should also return an error here if `k` doesn't have prefix `bgp.peers.`. Currently it allows setting arbitrary config keys like `bgp.abc.{peer}.holdtime`.

You can add the condition to the if statement below:

```go
if len(fields) != 4 || !strings.HasPrefix(k, "bgp.peers.") {
    return nil, fmt.Errorf("Invalid network configuration key: %q", k)
}
```

Reply:
Please quote the string with %q: `fmt.Errorf("Invalid network configuration key: %q", k)`

Reply:
Done and done